operator/v1/types_csi_cluster_driver.go

```diff
@@ -168,6 +168,61 @@ type AWSCSIDriverConfigSpec struct {
 	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov|aws-iso|aws-iso-b|aws-iso-e|aws-iso-f):kms:[a-z0-9-]+:[0-9]{12}:(key|alias)\/.*$`
 	// +optional
 	KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+
+	// efsVolumeMetrics sets the configuration for collecting metrics from EFS volumes used by the EFS CSI Driver.
+	// +optional
+	EFSVolumeMetrics *AWSEFSVolumeMetrics `json:"efsVolumeMetrics,omitempty"`
+}
+
+// AWSEFSVolumeMetricsState defines if the collection of volume metrics in the AWS EFS CSI Driver is enabled or disabled.
+// +openshift:enable:FeatureGate=AWSEFSDriverVolumeMetrics
+// +kubebuilder:validation:Enum=Enabled;Disabled
+type AWSEFSVolumeMetricsState string
+
+const (
+	// AWSEFSVolumeMetricsEnabled defines that the collection of volume metrics in the AWS EFS CSI Driver is enabled.
+	AWSEFSVolumeMetricsEnabled AWSEFSVolumeMetricsState = "Enabled"
```

Comment:
Consequence of enabling this options should be documented - it will do a recursive walk through the filesystem, which may take some CPU and memory on large volumes.
I think we should use `RecursiveWalk` to make it explicit instead of `Enabled`. This also opens opportunity for other mechanism, if EFS allows them in future.

Reply:
Added docs.
IMO `Enabled` offers more clarity. If the CSI offers other mechanisms in the future, we could add a `mode`-like field for this.
@JoelSpeed do you have any thoughts on this?

Reply:
I fall on Jan's side here, creating a mode field is not ideal as you then have a paired set of fields. The whole point of using enums is that we don't have to add additional paired fields

Reply:
I can change `Enabled/Disabled` to `RecursiveWalk/None`, but I'm not sure how to get rid of the "paired fields" (i.e., `Options` field).

Reply:
Done, PTAL